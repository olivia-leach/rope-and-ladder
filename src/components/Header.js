import React, { Component } from 'react';
import { Link } from "react-router-dom";
import SocialIcons from './SocialIcons';
import logo from '../logo.svg'
import "../styles/Header.scss";

class Header extends Component {
    render() {
        return <header className="main-header">
            <div className="container">
              <nav className="main-menu">
                {/*<ul>
                  <li><Link to="/">Home</Link></li>
                  <li><Link to="/tour">Tour</Link></li>
                  <li><Link to="/store">Store</Link></li>
                  <li><Link to="/contact">Contact</Link></li>
                </ul>*/}
              </nav>
              <nav className="main-menu social">
                <SocialIcons />
              </nav>
                {/*<div className='logo-container'>
                    <Link to="/">
                        <i className="fas fa-crown" />
                        {/*<img src={logo} alt="Rope and Ladder Logo" />
                    </Link>
              </div>*/}
            </div>
          </header>
    }
}

export default Header;
