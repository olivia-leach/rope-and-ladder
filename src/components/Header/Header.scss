$break-mobile: 500px;
$break-small: 767px;

.main-header {
    // position: absolute;
    width: 100%;
    top: 0;
    left: 0;
    z-index: 100;
    font-size: .9rem;
    position: fixed;
    background-color: transparent;
    transition: 0.5s cubic-bezier(0.19, 1, 0.22, 1) background-color;

    &.is-dark {
      background-color: #1F282C;
      padding-bottom: 1.5rem;
      box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
    }
}

a {
    transition: 0.2s ease-in-out all;
    color: #fff;
    text-decoration: none;
    opacity: .8;

    &:hover {
        opacity: 1;
    }
}

.main-header {
    font-weight: bold;
    letter-spacing: 2px;
}

.social-icon {
    color: #fff;
}

ul {
    margin: 0;
}

li {
    list-style: none;
    padding: 0;
}

.main-menu {
    margin-left: 6%;
    padding-top: 1rem;

    a {
        transition: 0.2s ease-in-out all;
        position: relative;
        padding: 0.5rem;
        letter-spacing: 2px;

        &:after {
            transition: 0.2s ease-in-out all;
            content: '';
            position: absolute;
            bottom: -5px;
            left: -1px;
            width: 0%;
            height: 2px;
            background-color: #fff;
            border-radius: 1px;
        }

        &.active {
            opacity: 1;
            pointer-events: none;
            letter-spacing: 4px;

            &:after {
                content: '';
                width: 100%;
            }
        }
    }
}

.social-menu {
    padding-top: 1rem;
    float: right;
    margin-right: 10%;

    @media screen and (max-width: $break-mobile) {
        margin-right: 1rem;
        padding-top: 0;
      }
}

.main-menu {
    display: inline-block;

    a {
      cursor: pointer;
    }

    @media screen and (max-width: $break-mobile) {
      display: none;
    }
}

.main-menu ul, .main-menu li {
    display: inline-block;
    padding: 0;
}

.main-menu li {
    padding: 1rem;
    text-transform: uppercase;
}

.logo-container {
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
    margin-top: 1.2rem;
    top: 0;
}

.logo-container a {
    color: #fff;
    font-size: 2rem;
}

.header-logo-container {
    width: 100%;
    display: flex;
    justify-content: center;

    @media screen and (max-width: $break-small) {
        margin-top: 1.5rem;
    }
  }
  
  .vertical-nav {
    position: absolute;
    left: 0;
    top: 0;
    z-index: 50;
    display: none;
    z-index: 200;

    @media screen and (max-width: $break-mobile) {
      display: block;
    }
  
    .menu-button {
      padding: 1.5rem 1rem;
      font-size: 1rem;
      background-color: #2c393e;
      border: 0;
      color: #fff;
      margin-left: 15px;
      position: absolute;
      top: 0;
      cursor: pointer;
      opacity: 0.8;
  
      &:hover {
        opacity: 1;
      }
  
      &.hide {
        opacity: 0;
      }
    }
  }
  
  .nav-open {
    transition: 0.2s cubic-bezier(.25,.8,.25,1) transform;
    background-color: rgba(#2c393e, 0.8);
    color: #fff;
    box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
    transform: translateY(0);
    padding: 0;
    margin-left: 15px;
    display: flex;
    align-items: flex-start;
    flex-direction: column;
  
    .menu-button {
      margin-left: 0;
      background-color: transparent;
      position: relative;
      flex-shrink: 0;
    }
  
    &.hide {
      transform: translateY(-100%);
      box-shadow: none;
    }
  
    .nav-item {
      transition: 0.2s ease-in-out background-color;
      text-transform: uppercase;
      letter-spacing: 2px;

      a {
        opacity: 1;
        display: block;
        padding: 1rem 2rem;
        background-color: rgba(#2c393e, 0);

        &:hover {
          background-color: rgba(#2c393e, 1);
        }
      }
  
      &:last-of-type a {
        padding-bottom: 2rem;
      }
    }
  }