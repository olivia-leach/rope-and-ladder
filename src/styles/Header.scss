.main-header {
    padding-top: 3rem;
    position: fixed;
    width: 100%;
    top: 0;
    left: 0;
    z-index: 5;
    font-size: .9rem;
}

a {
    color: #fff;
    text-decoration: none;
}

.main-header a {
    font-weight: bold;
    letter-spacing: 1px;
}

.social-icon {
    color: #fff;
}

li {
    list-style: none;
}

.main-menu.social {
    float: right;
    margin-right: 4rem;
    display: none;
}

.main-menu {
    display: inline-block;
}

.main-menu ul, .main-menu li {
    display: inline-block;
    padding: 0;
}

.main-menu li {
    padding: 1rem;
    text-transform: uppercase;
}

.container {
    margin: 0 auto;
    position: relative;
}

.logo-container {
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
    margin-top: 1.2rem;
    top: 0;
}

.logo-container a {
    color: #fff;
    font-size: 2rem;
}
