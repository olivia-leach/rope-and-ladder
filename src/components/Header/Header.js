import React, { Component, Fragment } from 'react';
import { Link } from 'react-scroll'
import SocialIcons from '../SocialIcons';

import './Header.scss'

const pages = ['home', 'store', 'about', 'listen']

class Header extends Component {
    constructor(props) {
      super(props)
      const hash = props.history.location.hash
      this.state = {
        showNav: false,
        isHeaderDark: hash && hash.substr(1),
      }
    }

    toggleNav = () => {
      this.setState(prevState => ({
        showNav: !prevState.showNav
      }))
    }

    handleOnSetActive = (route) => {
      this.setState({ isHeaderDark: route !== 'home'})
    }

    render() {
        return (
          <Fragment>
          <div className='vertical-nav'>
            <button onClick={this.toggleNav} className={`menu-button ${this.state.showNav ? 'hide' : ''}`}>
              {this.state.showNav ? <i className='fa fa-times' /> : <i className='fa fa-bars' />}
            </button>
            <div className={`nav-open ${!this.state.showNav ? 'hide' : ''}`}>
              <button onClick={this.toggleNav} className='menu-button'>
                {this.state.showNav ? <i className='fa fa-times' /> : <i className='fa fa-bars' />}
              </button>
              <div className='vertical-nav-links'>
                {pages.map(page => <div key={page} className='nav-item'><Link to={page} onClick={this.toggleNav}>{page}</Link></div>)}
              </div>
            </div>
          </div>
            <header className={`main-header ${this.state.isHeaderDark ? 'is-dark' : ''} ${window.scrollY}`}>
            <div className="container">
              <nav className="main-menu">
                <ul>
                  {pages.map(page => <li key={page}><Link to={page} offset={-90} hashSpy={page !== 'home'} spy onSetActive={this.handleOnSetActive}>{page}</Link></li>)}
                </ul>
              </nav>
              <nav className="social-menu">
                <SocialIcons />
              </nav>
            </div>
          </header>
          </Fragment>
        )
    }
}

export default Header;
