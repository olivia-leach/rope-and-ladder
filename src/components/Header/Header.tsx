import React, { useCallback, useState } from 'react'
import { Link } from 'react-scroll'
import { useHistory, useLocation } from 'react-router-dom'
import { NavHashLink } from 'react-router-hash-link'

import SocialIcons from '../SocialIcons'

import './Header.scss'

const pages = ['home', 'store', 'about', 'listen']

const Header = () => {
  const location = useLocation()
  const { hash } = location
  const [showNav, setShowNav] = useState(false)
  const [isHeaderDark, setIsHeaderDark] = useState(hash?.substr(1) || false)

  const toggleNav = useCallback(() => {
    setShowNav(!showNav)
  }, [showNav])

  const handleOnSetActive = useCallback((route: string) => {
    setIsHeaderDark(route !== 'home')
  }, [])

  return (
    <>
      <div className="vertical-nav">
        <button
          onClick={toggleNav}
          className={`menu-button ${showNav ? 'hide' : ''}`}
        >
          {showNav ? (
            <i className="fa fa-times" />
          ) : (
            <i className="fa fa-bars" />
          )}
        </button>
        <div className={`nav-open ${!showNav ? 'hide' : ''}`}>
          <button onClick={toggleNav} className="menu-button">
            {showNav ? (
              <i className="fa fa-times" />
            ) : (
              <i className="fa fa-bars" />
            )}
          </button>
          <div className="vertical-nav-links">
            {pages.map(page => (
              <div key={page} className="nav-item">
                {location.pathname === '/' ? (
                  <Link to={page} onClick={toggleNav}>
                    {page}
                  </Link>
                ) : (
                  <NavHashLink onClick={toggleNav} to={`/#${page}`}>
                    {page}
                  </NavHashLink>
                )}
              </div>
            ))}
          </div>
        </div>
      </div>
      <header
        className={`main-header ${isHeaderDark ? 'is-dark' : ''} ${
          window.scrollY
        }`}
      >
        <div className="container">
          <nav className="main-menu">
            <ul>
              {location.pathname === '/'
                ? pages.map(page => (
                    <li key={page}>
                      <Link
                        to={page}
                        offset={-90}
                        hashSpy={page !== 'home'}
                        spy
                        onSetActive={handleOnSetActive}
                      >
                        {page}
                      </Link>
                    </li>
                  ))
                : pages.map(page => (
                    <li key={page}>
                      <NavHashLink to={`/#${page}`}>{page}</NavHashLink>
                    </li>
                  ))}
            </ul>
          </nav>
          <nav className="social-menu">
            <SocialIcons />
          </nav>
        </div>
      </header>
    </>
  )
}

export default Header
