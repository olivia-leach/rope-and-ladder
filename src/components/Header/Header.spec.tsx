import { render, fireEvent } from '@testing-library/react'
import React from 'react'
import { StaticRouter } from 'react-router'
import Header from './Header'

const component = (
  <StaticRouter>
    <Header />
  </StaticRouter>
)

describe('Header', () => {
  it('should match snapshot', () => {
    const { container } = render(component)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot when nav is active', () => {
    const { container, getByTestId } = render(component)
    fireEvent.click(getByTestId('menuButton'))
    expect(container).toMatchSnapshot()
  })
})
