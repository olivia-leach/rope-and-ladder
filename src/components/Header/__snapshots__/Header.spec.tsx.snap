// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div>
  <div
    class="vertical-nav"
  >
    <button
      class="menu-button "
      data-testid="menuButton"
    >
      <i
        class="fa fa-bars"
      />
    </button>
    <div
      class="nav-open hide"
    >
      <button
        class="menu-button"
      >
        <i
          class="fa fa-bars"
        />
      </button>
      <div
        class="vertical-nav-links"
      >
        <div
          class="nav-item"
        >
          <a>
            home
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a>
            store
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a>
            about
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a>
            listen
          </a>
        </div>
      </div>
    </div>
  </div>
  <header
    class="main-header  0"
  >
    <div
      class="container"
    >
      <nav
        class="main-menu"
      >
        <ul>
          <li>
            <a>
              home
            </a>
          </li>
          <li>
            <a>
              store
            </a>
          </li>
          <li>
            <a>
              about
            </a>
          </li>
          <li>
            <a>
              listen
            </a>
          </li>
        </ul>
      </nav>
      <nav
        class="social-menu"
      >
        <ul
          class="social-list"
        >
          <li>
            <a
              class="social-icon"
              href="https://www.instagram.com/RopeAndLaddermusic/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-instagram"
              />
            </a>
          </li>
          <li>
            <a
              class="social-icon"
              href="https://www.twitter.com/Rope_and_Ladder"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-twitter"
              />
            </a>
          </li>
          <li>
            <a
              class="social-icon"
              href="https://www.facebook.com/RopeAndLadder"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-facebook"
              />
            </a>
          </li>
          <li>
            <a
              class="social-icon"
              href="https://www.youtube.com/channel/UChM5AgIbOSZssaG8N9BJ00w"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-youtube-square"
              />
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</div>
`;

exports[`Header should match snapshot when nav is active 1`] = `
<div>
  <div
    class="vertical-nav"
  >
    <button
      class="menu-button hide"
      data-testid="menuButton"
    >
      <i
        class="fa fa-times"
      />
    </button>
    <div
      class="nav-open "
    >
      <button
        class="menu-button"
      >
        <i
          class="fa fa-times"
        />
      </button>
      <div
        class="vertical-nav-links"
      >
        <div
          class="nav-item"
        >
          <a>
            home
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a>
            store
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a>
            about
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a>
            listen
          </a>
        </div>
      </div>
    </div>
  </div>
  <header
    class="main-header  0"
  >
    <div
      class="container"
    >
      <nav
        class="main-menu"
      >
        <ul>
          <li>
            <a>
              home
            </a>
          </li>
          <li>
            <a>
              store
            </a>
          </li>
          <li>
            <a>
              about
            </a>
          </li>
          <li>
            <a>
              listen
            </a>
          </li>
        </ul>
      </nav>
      <nav
        class="social-menu"
      >
        <ul
          class="social-list"
        >
          <li>
            <a
              class="social-icon"
              href="https://www.instagram.com/RopeAndLaddermusic/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-instagram"
              />
            </a>
          </li>
          <li>
            <a
              class="social-icon"
              href="https://www.twitter.com/Rope_and_Ladder"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-twitter"
              />
            </a>
          </li>
          <li>
            <a
              class="social-icon"
              href="https://www.facebook.com/RopeAndLadder"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-facebook"
              />
            </a>
          </li>
          <li>
            <a
              class="social-icon"
              href="https://www.youtube.com/channel/UChM5AgIbOSZssaG8N9BJ00w"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-youtube-square"
              />
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</div>
`;
